docs/performance/inserts/methods.txt

```diff
@@ -0,0 +1,283 @@
+.. _insert_methods:
+
+==============
+Insert Methods
+==============
+
+There are multiple ways to insert data into a CrateDB database.
+
+Some insert methods can be faster than others, depending on your setup.
+Choosing the best insert method is an easy way to improve insert performance.
+
+.. rubric:: Table of Contents
+
+.. contents::
+   :local:
+
+.. _insert_statement_types:
+
+Statement Types
+===============
+
+There are three types of insert statement:
+
+- :ref:`Single inserts <inserts_single_inserts>`
+- :ref:`UNNEST <inserts_unnest>`
+- :ref:`Multiple value expressions <inserts_multiple_values>`
+
+Each type of statement can be issued using any of the three types of
+:ref:`client approaches <inserts_client_approaches>`, which we cover in
+the next section.
+
+.. _inserts_single_inserts:
+
+Single Inserts
+--------------
+
+`Single inserts`_ are the most basic sort of insert statement, and look like
+this:
+
+.. code-block:: psql
+
+    cr> INSERT INTO my_table (column_a) VALUES ("value 1");
+    INSERT OK, 1 row affected  (... sec)
+
+Single inserts are typically very fast with CrateDB. A small cluster can
+easily handle several thousand inserts per second.
+
+However, it's important to note that:
+
+ 1. Every insert is first applied to the primary shard
+
+ 2. After the primary shard has been updated, the insert is then individually
+    communicated in parallel to every configured replica shard
+
+ 3. CrateDB will not return a response until all replica shards have been
+    updated
+
+The overhead for each insert (query parsing, planning, and execution, which
+includes network connection setups, connection teardowns, and connection
```

Comment:
Not sure if we should include connection setups and teardown here. Node-to-node the connections are kept alive. Similar with HTTP which supports keep-alive. Postgres connection is also kept open until the client disconnects.

Reply:
changed to:

> The overhead for each insert (query parsing, planning, and execution, and
potentially network connection overhead too) starts to add up for very heavy
insert workloads.